Frontend/src/api/auth/get_jwt.ts

```diff
@@ -0,0 +1,29 @@
+import { ErrorResponse } from '../types'
+
+export interface SuccessfulResponse {
+  token: string
+  error: false
+}
+
+export async function getJWT(
+  reauthenticate = false
+): Promise<ErrorResponse | SuccessfulResponse> {
+  // cache the jwt token, if it has expired we just need to reauthenticate
+  if (reauthenticate || localStorage.getItem('jwt') === null) {
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore AUTH_URL is injected at build time
+      const response = await fetch(`${process.env.AUTH_URL}`)
+      const token = response.headers.get('authorization')
+      if (response.ok && token !== null) {
```

Comment:
First check whether the response is OK, _then_ access headers

Reply:
needed to nest the ifs, but definitely better than before